### PR TITLE
Added new methodology to fins the test run time distributions

### LINKED ISSUE #13: Update the statistical tests and test distributions
The old methodology used in OSP model is intricate and there are issues like #11 using the scipy library so approaching it a bit differently with a different wrapper library for scipy that helps resolve the issue and makes easier to understand and monitor.

### PR DESCRIPTION
**Pull request by aakankshaduggal**
Added new methodology to fins the test run time distributions and also find the best distributions for failing and passing data sets. Next steps : Calculate OSP. Also added old OSP helper function for reference.

Resolves #13 #11

### CODE REVIEW

File: notebooks/test_distributions.ipynb [hunk 1]
```diff
@@ -0,0 +1,975 @@
+{
```
Comment (oindrillac) [inline notebooks/test_distributions.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    passing_df = pd.read_csv('../data/processed/passing_test.csv')
</pre><p>Can we use the data from the previous data collection step? Or if we dont want to store data locally, and if it doesnt make the notebook long, maybe we can store in-memory and do it in a single notebook?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/redhat-et/optimal-stopping-point-tool/pull/14/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (oindrillac) [inline notebooks/test_distributions.ipynb:1]: <p>The failing distribution seems heavily skewed and Im not sure we will get a suitable fit without a transformation. Can we try log transform, or sq root transform or box-cox on the data here and see what gets us easier fits?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/redhat-et/optimal-stopping-point-tool/pull/14/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (oindrillac) [inline notebooks/test_distributions.ipynb:1]: <p>Why are we limiting to only common distributions? Can we try it over all the distributions to see which offers the the lowest sum square error?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/redhat-et/optimal-stopping-point-tool/pull/14/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (oindrillac) [reply notebooks/test_distributions.ipynb:1]: <p>Out of all the tests, we collect from step 1, I think we can leave out about 20% to evaluate our approach on and fit the rest to the curves. WDYT?</p>

Comment (aakankshaduggal) [reply notebooks/test_distributions.ipynb:1]: sure, that makes sense. We can combine this, and I can append this to the same notebook. 
And can split 80-20 and use that for validation

Comment (aakankshaduggal) [reply notebooks/test_distributions.ipynb:1]: Sure, that's something I can look at.

Comment (aakankshaduggal) [reply notebooks/test_distributions.ipynb:1]: Yes, can add that here.

Comment (aakankshaduggal) [reply notebooks/test_distributions.ipynb:1]: The new updates are making the notebook super long and kernel is also dying, so limiting this to two notebooks atm. We can definitely make it one notebook once we are pass the experimenting phase.

**Review by oindrillac — APPROVED** [review]
Lgtm! thanks for the changes

**Pull request merged**